File: 프로그래머스/2/42885. 구명보트/구명보트.py
def solution(people, limit):
    answer = 0
    p = sorted(people,reverse=True)
    l = 0
    r = len(p)-1
    
    while r>l :
        if p[l] + p[r] <= limit:
            l +=1
            r -=1
            answer +=1
            if l>r:
                break
            elif l==r:
                answer +=1
                break
        else:
            l+=1
            answer +=1
            if l==r:
                answer+=1
                break
    return answer

File: 프로그래머스/2/42885. 구명보트/구명보트.js
function solution(people, limit) {
    let count = 0;
    people.sort((a, b) => a - b);
    
    while (people.length) {
        if (people[0] + people.at(-1) <= limit) {
            people.shift();
            people.pop();
            count++;
        } else {
            people.pop();
            count++;
        }
    }
    return count;
}